extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/OwBaseBridgeHandler.java

```diff
@@ -0,0 +1,162 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.onewire.internal.handler;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.binding.onewire.internal.OwException;
+import org.eclipse.smarthome.binding.onewire.internal.OwPageBuffer;
+import org.eclipse.smarthome.binding.onewire.internal.device.OwDeviceParameter;
+import org.eclipse.smarthome.binding.onewire.internal.device.OwSensorType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OwBaseBridgeHandler} class implements the refresher and the interface for reading from the bridge
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public abstract class OwBaseBridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(OwBaseBridgeHandler.class);
+
+    // set by implementation when bridge is ready
+    protected boolean refreshable = false;
+
+    public OwBaseBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    protected Runnable owRefresher = () -> {
+        try {
+            if (isRefreshable()) {
+                long now = System.currentTimeMillis();
+                List<Thing> thingList = getThing().getThings();
+                int thingCount = thingList.size();
+                logger.trace("called owRefresher at {}, {} childs", now, thingCount);
+
+                Iterator<Thing> childListIterator = thingList.iterator();
+                while (childListIterator.hasNext()) {
+                    Thing owThing = childListIterator.next();
+
+                    logger.trace("getting handler for {} ({} to go)", owThing.getLabel(), thingCount);
+                    OwBaseThingHandler owHandler = (OwBaseThingHandler) owThing.getHandler();
+                    if (owHandler != null) {
+                        if (owHandler.isRefreshable()) {
+                            logger.trace("{} initialized, refreshing", owThing.getLabel());
+                            owHandler.refresh(OwBaseBridgeHandler.this, now);
+                        } else {
+                            logger.trace("{} not initialized, skipping refresh", owThing.getLabel());
+                        }
+                    } else {
+                        logger.debug("{} handler missing", owThing.getLabel());
+                    }
+                    thingCount--;
+                }
+            }
+        } catch (RuntimeException e) {
```

Comment:
Please do not catch `RuntimeException` directly, this will also accidentally catch `NullpointerException`s.... Please catch specific ones or throw properly typed Exceptions, which is even better...

Reply:
I removed the check. It never happened again after I changed the OwException handling.